lib/datadog/core/telemetry/logger.rb

```diff
@@ -25,7 +25,17 @@ def error(description)
           private
 
           def instance
-            Datadog.send(:components).telemetry
+            # `allow_initialization: false` to prevent deadlock from components lifecycle
+            components = Datadog.send(:components, allow_initialization: false)
```

Comment:
The reason for this deadlock is quite subtle; I would suggest writing a bit more here to explain what's going on and why we're doing this (so in the future other folks don't need to "reverse engineer" what happened here)